docs/config-validation.md

```diff
@@ -0,0 +1,183 @@
+# Configuration Validation
+
+We use a fork of https://github.com/go-playground/validator which can be found
+at https://github.com/letsencrypt/validator. 
+
+## Usage
+
+A `validate` subcommand has been included in the `boulder` binary. You can check
+the usage with `boulder validate -help`.
+
+Use the following syntax to validate a config file:
+```shell
+boulder validate -component <component> -config <config file>
+```
+
+For instance, to validate the `boulder-ca` config file you can run:
+
+```shell
+boulder validate -component boulder-ca -config test/config/ca.json`
+```
+
+For a complete list of `boulder` components which support config validation you
+can run:
+```shell
+boulder validate -list`
+```
+
+## Struct Tag Tips
+
+You can find the full list of struct tags supported by the validator [here]
+(https://pkg.go.dev/github.com/go-playground/validator/v10#section-documentation).
+The following are some tips for struct tags that are commonly used in our
+configuration files.
+
+### `required`
+
+The required tag means that the field is not allowed to take its zero value, or
+equivalently, is not allowed to be omitted. Note that this does not validate
+that slices or maps have contents, it simply guarantees that they are not nil.
+For fields of those types, you should use min=1 or similar to ensure they are
+not empty.
+
+There are also "conditional" required tags, such as `required_with`,
+`required_with_all`, `required_without`, `required_without_all`, and
+`required_unless`. These behave exactly like the basic required tag, but only if
+their conditional (usually the presence or absence of one or more other named
+fields) is met.
+
+### `omitempty`
+
+The omitempty tag allows a field to be empty, or equivalently, to take its zero
+value. If the field is omitted, none of the other validation tags on the field
+will be enforced. This can be useful for tags like validate="omitempty,url", for
+a field which is optional, but must be a URL if it is present.
+
+The omitempty tag can be "overruled" by the various conditional required tags.
+For example, a field with tag `validate="omitempty,url,required_with=Foo"` is
+allowed to be empty when field Foo is not present, but if field Foo is present,
+then this field must be present and must be a URL.
+
+### `-`
+Normally, config validation descends into all struct-type fields, recursively
```

Comment:
nit: missing empty line between section header and body. Here and a few other places below.
```suggestion
### `-`

Normally, config validation descends into all struct-type fields, recursively
```